### PR TITLE
Example.py: allow CoAP port customization and debug logging level

### PR DESCRIPTION
**Pull request by chemelli74**
2 new parameters for example.py:

- "`--socket_port`" : allows CoAP port customization.
- "`--debug`": set debug logging level.

### CODE REVIEW

**Comment by thecode** [comment]
I find it very odd that we know have two places for setting the port, one in the constructor and one in the `initialize` function.

The problem was that from the first we should have port the port in the constructor.

Consider this code:
```python
coap = COAP(1234)
coap.initialize(1111)
```
We have now speficied two different ports and the port from the constructor will override the port  set by `initialize`

I think we should consider one of the following:
1. Move the port to the constructor and remove it from `initialize`, clean solution,  **breaking change**
2. Modify the example to pass the port via `initialize` (without using asynchronous context manager) - simple change

File: example.py [hunk 1]
```diff
@@ -216,6 +225,9 @@ def get_arguments() -> Tuple[argparse.ArgumentParser, argparse.Namespace]:
     parser.add_argument(
         "--gen2", "-g2", action="store_true", help="Force Gen 2 (RPC) device"
     )
+    parser.add_argument(
+        "--debugger", "-deb", action="store_true", help="Enable debug level for logging"
```
Comment (thecode) [inline example.py:229]: This not really enable a debugger, just set a debug log, so it would be better to call it `--debug`

**Comment by chemelli74** [comment]
> I find it very odd that we know have two places for setting the port, one in the constructor and one in the `initialize` function.

Agree

> I think we should consider one of the following:
> 
> 1. Move the port to the constructor and remove it from `initialize`, clean solution,  **breaking change**
> 2. Modify the example to pass the port via `initialize` (without using asynchronous context manager) - simple change

I prefer the second option, so I implemented it ;-)

Simone

File: example.py [hunk 2]
```diff
@@ -20,6 +21,13 @@
 from aioshelly.rpc_device import RpcDevice
 
 
+async def get_coap_context(port: int | None = None) -> COAP:
```
Comment (thecode) [inline example.py:24]:
```suggestion
async def get_coap_context(port: int) -> COAP:
```

File: example.py [hunk 3]
```diff
@@ -75,19 +89,24 @@ async def test_single(
...
+async def test_devices(init: bool, timeout: float, port: int, gen: int | None) -> None:
     """Test multiple devices."""
     device_options = []
     with open("devices.json") as fp:
         for line in fp:
             device_options.append(ConnectionOptions(**json.loads(line)))
 
-    async with aiohttp.ClientSession() as aiohttp_session, COAP() as coap_context:
+    async with aiohttp.ClientSession() as aiohttp_session:
         results = await asyncio.gather(
```
Comment (thecode) [inline example.py:100]:
```suggestion
        coap_context = await get_coap_context(port)
        results = await asyncio.gather(
```

File: example.py [hunk 4]
```diff
@@ -75,19 +89,24 @@ async def test_single(
...
             *[
                 asyncio.wait_for(
                     connect_and_print_device(
-                        aiohttp_session, coap_context, options, init, timeout, gen
+                        aiohttp_session,
+                        await get_coap_context(port),
```
Comment (thecode) [inline example.py:105]:
```suggestion
                        coap_context,
```

**Review by thecode — APPROVED** [review]

**Pull request merged**